File: DevReviewFrontend/src/types/officeHour.ts
import type { OfficeHourStatus } from './enums'

export interface OfficeHour {
  id: string
  mentorId: string
  startTime: string
  durationMinutes: number
  endTime: string
  topic: string
  price: number | null
  status: OfficeHourStatus
  bookedById: string | null
  bookingDescription: string | null
}

export interface OfficeHourDetail extends OfficeHour {
  mentorName: string
  mentorEmail: string
  bookedByName: string | null
  bookedByEmail: string | null
  mentorImpression: string | null
  authorImpression: string | null
}

export interface CreateOfficeHourPayload {
  startTime: string
  durationMinutes: number
  topic: string
  price?: number | null
}

export interface BookOfficeHourPayload {
  bookingDescription?: string
}

export interface CancelOfficeHourPayload {
  cancellationReason?: string
}

export interface SubmitOfficeHourFeedbackPayload {
  impression: string
}

File: API/DevReview.Domain/Entities/OfficeHour.cs
using System;
using DevReview.Domain.Enums;

namespace DevReview.Domain.Entities
{
    public class OfficeHour
    {
        public Guid Id { get; set; }
        public Guid MentorId { get; set; }
        public User? Mentor { get; set; }
        public DateTime StartTime { get; set; }
        public int DurationMinutes { get; set; }
        public DateTime EndTime { get; set; }
        public string Topic { get; set; } = string.Empty;
        public decimal? Price { get; set; }
        public OfficeHourStatus Status { get; set; } = OfficeHourStatus.Available;
        public Guid? BookedById { get; set; }
        public User? BookedBy { get; set; }
        public string? BookingDescription { get; set; }
        public string? MentorImpression { get; set; }
        public string? AuthorImpression { get; set; }
    }
}
